src/Lecture2.hs

```diff
@@ -264,7 +331,16 @@ data EvalError
 It returns either a successful evaluation result or an error.
 -}
 eval :: Variables -> Expr -> Either EvalError Int
-eval = error "TODO"
+eval _ (Lit i) = Right i
+eval symbolTable (Var symbol) = case lookup symbol symbolTable of
+    Nothing  -> Left (VariableNotFound symbol)
+    Just value -> Right value
+eval symbolTable (Add e1 e2) = addExpr (eval symbolTable e1) (eval symbolTable e2)
+  where
+    addExpr :: Either EvalError Int -> Either EvalError Int -> Either EvalError Int
+    addExpr (Left x) _ = Left x
+    addExpr _ (Left x) = Left x
+    addExpr (Right x) (Right y) = Right (x + y)
```

Comment:
That's a great helper function! 👏🏻 
Once you learn about Monads, you'll be able to reduce this function to a single line ✂️

Reply:
Thanks!